### ISSUE TITLE
undefined-variable false positives with async/await

### ISSUE THREAD

**Issue by belak** [PyCQA/pylint#767]
https://travis-ci.org/belak/python-seabird/builds/100674977

https://github.com/belak/python-seabird/blob/async-metar/seabird/plugins/metar.py

pylint dep versions:
astroid-1.4.3 colorama-0.3.5 lazy-object-proxy-1.2.1 pylint-1.5.2 wrapt-1.10.6

It also errors with pylint==1.4.5, but I think it ends up being astroid errors there.

The relevant portion of the code for the lazy:

```
    async def metar_callback(self, cmd):
        resp = await aiohttp.get(METAR_URL.format(cmd.remainder.upper()))
        if resp.status == 404:
            cmd.mention_reply('{} is not a valid METAR code'.format(
                cmd.remainder))
            return
        elif resp.status != 200:
            cmd.mention_reply('Unable to get METAR for {}'.format(
                cmd.remainder))
            return

        try:
            data = await resp.text()
            cmd.mention_reply(data.split('\n')[1])
        except KeyError:
            cmd.mention_reply('Malformed METAR data for {}'.format(
                cmd.remainder))
```

And the log:
```
************* Module seabird.plugins.metar
E: 21,50: Undefined variable 'cmd' (undefined-variable)
E: 22,11: Undefined variable 'resp' (undefined-variable)
E: 23,12: Undefined variable 'cmd' (undefined-variable)
E: 24,16: Undefined variable 'cmd' (undefined-variable)
E: 26,13: Undefined variable 'resp' (undefined-variable)
E: 27,12: Undefined variable 'cmd' (undefined-variable)
E: 28,16: Undefined variable 'cmd' (undefined-variable)
E: 32,25: Undefined variable 'resp' (undefined-variable)
E: 33,12: Undefined variable 'cmd' (undefined-variable)
E: 33,30: Undefined variable 'data' (undefined-variable)
E: 35,12: Undefined variable 'cmd' (undefined-variable)
E: 36,16: Undefined variable 'cmd' (undefined-variable)
```

**Comment by PCManticore** [comment]
Thanks for the report. The fix will be released as part of 1.5.3 in the following days (I hope on Monday).
